### PR TITLE
Add curl.md

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**
Add the curl document providing example console inputs used to communicate with the reSolve HTTP API

### CODE REVIEW

File: docs/basics/curl.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+This document demonstrates how to communicate with a reSolve backend using the HTTP API. To try the provided console inputs on your machine, download and run the [Shopping List](https://github.com/reimagined/resolve/tree/master/examples/shopping-list) example project.
```
Comment (dirk-pieterse) [inline docs/basics/curl.md]: "This document demonstrates how to communicate with a reSolve backend using the HTTP API." -> This document demonstrates how to use the HTTP API to communicate with a reSolve backend.

a reSolve **backend**?

Comment (dirk-pieterse) [reply docs/basics/curl.md]: "To try the provided console inputs on your machine" to try?

File: docs/basics/curl.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+1. Create a new shopping list name "List 1":
```
Comment (dirk-pieterse) [inline docs/basics/curl.md]: name "List 1"? name?

File: docs/basics/curl.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Content-Type: text/html; charset=utf-8
+Content-Length: 140
+ETag: W/"8c-rWsIpzFOfkV3y9g6x9FlenTaG/A"
+Date: Tue, 02 Oct 2018 12:17:57 GMT
+Connection: keep-alive
+
+{"id":"12345-new-shopping-list","name":"List 1","list":[{"id":"1","text":"Beer","checked":false},{"id":"2","text":"Chips","checked":false}]}
+```
+
+Below you can see the newly created list and its items on a page.
```
Comment (dirk-pieterse) [inline docs/basics/curl.md]: on a page? Which page?

File: docs/basics/curl.md (hunk 2 above)
Comment (EugeniyBurmistrov) [reply docs/basics/curl.md]: Changed to 'named'

File: docs/basics/curl.md (hunk 1 above)
Comment (EugeniyBurmistrov) [reply docs/basics/curl.md]: > "To try the provided console inputs on your machine" to try?

Changed to "To **test** the provided console inputs on your machine"

File: docs/basics/curl.md (hunk 3 above)
Comment (EugeniyBurmistrov) [reply docs/basics/curl.md]: Changed to "on the Shopping List application's page."

File: docs/basics/curl.md (hunk 1 above)
Comment (dirk-pieterse) [reply docs/basics/curl.md]: Do not use 'using'. Rewrite.

File: docs/basics/curl.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
+This document demonstrates how to use the HTTP API to communicate with a reSolve backend. To try the provided console inputs on your machine, download and run the [Shopping List](https://github.com/reimagined/resolve/tree/master/examples/shopping-list) example project.
```
Comment (dirk-pieterse) [inline docs/basics/curl.md]: To try -> To test

File: docs/basics/curl.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+Content-Type: text/html; charset=utf-8
+Content-Length: 58
+ETag: W/"3a-jyqRShDvCZnc9uCOPi31BlQFznA"
+Date: Tue, 02 Oct 2018 12:11:43 GMT
+Connection: keep-alive
+
+{"id":"12345-new-shopping-list","name":"List 1","list":[]}
+```
+
+3. Create an item in the shopping list:
```
Comment (dirk-pieterse) [inline docs/basics/curl.md]: Create an item -> Add an item?

**Review by dirk-pieterse — APPROVED** [review]

**Pull request merged**